kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -767,19 +766,15 @@ impl<'tcx> GotocCtx<'tcx> {
 
         // Then we check if the type allows "raw" initialization for the cases
         // where memory is zero-initialized or entirely uninitialized
-        if intrinsic == "assert_zero_valid"
-            && !layout.might_permit_raw_init(self, InitKind::Zero, false)
-        {
+        if intrinsic == "assert_zero_valid" && !self.tcx.permits_zero_init(layout) {
```

Comment:
Relevant to #1273.

Reply:
Sounds we close that?

Reply:
I want to take a look at other backends to confirm we aren't missing anything here. Then I'll close if that's the case.